### ISSUE TITLE
mypy issues with DeclarativeBase

### ISSUE THREAD

**Issue by YaraslauZhylko** [DanCardin/sqlalchemy-declarative-extensions#37]
`sqlalchemy-declarative-extensions` documentation suggests to declare the ORM base class like this:
```python
from sqlalchemy_declarative_extensions.sqlalchemy import declarative_base
from sqlalchemy_declarative_extensions import declarative_database

_Base = declarative_base()

@declarative_database
class Base(_Base):
    __abstract__ = True
```

But as of now, `declarative_base` [is being deprecated](https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.declarative_base):
> Changed in version 2.0: Note that the [declarative_base()](https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.declarative_base) function is superseded by the new [DeclarativeBase](https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.DeclarativeBase) class, which generates a new “base” class using subclassing, rather than return value of a function. This allows an approach that is compatible with [PEP 484](https://peps.python.org/pep-0484/) typing tools.

So I try to use `DeclarativeBase` as a base class like this:
```python
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy_declarative_extensions import declarative_database

@declarative_database
class Base(DeclarativeBase):
    __abstract__ = True
```
And as a result, I get a `mypy` error:
> error: Value of type variable "T" of "declarative_database" cannot be "type[Base]"  [type-var]

I guess the following line needs some adjustment to get work with both the old and the new declaration style:
 https://github.com/DanCardin/sqlalchemy-declarative-extensions/blob/bd52ff8bd90220c97a3a20b62dba4d37857b1dfb/src/sqlalchemy_declarative_extensions/api.py#L24

**Comment by DanCardin** [comment]
Before I release it, I wonder if you might install this branch https://github.com/DanCardin/sqlalchemy-declarative-extensions/pull/38 and see if it fixes your issue?

I dont have any real sqlalchemy2 projects handy, and I'm not entirely convinced this wont have side effects.

**Comment by DanCardin** [comment]
I've reverted to runtime checks here, because it seems to be impossible/impractical to have actually typed input types that support both versions.

As a result, I suspect it's much more likely to "just work" so I just merged/released it in v0.6.4.

Lmk if this does/doesnt solve your problem!

**Comment by YaraslauZhylko** [comment]
I'm sorry I missed the GitHub notification.

I tried the `0.6.4` release and it seems to have fixed the issue. At least I do not need `# type: ignore` any more. 😅 

No regressions encountered so far.
